### PR TITLE
Find the proper best candidate when the focus moves inside the iframe

### PR DESCRIPTION
**Pull request by jihyerish**
* Issue: Focus always goes to the top-most element when it moves inside the iframe not considering the scroll offset of the iframe)
* Solution: 
   1) Get the rect of iframe based on its containing document
   2) Set the iframe's position as (0,0) because the rects of elements inside the iframe don't know the real iframe's position.
* Added more test case for checking this update

### CODE REVIEW

File: tests/internal/iframe-test.html [hunk 1]
```diff
@@ -10,11 +10,11 @@
   <script src="test.js"></script>
 </head>
 <body onload="onload()">
-	<iframe src="iframe_sub.html" id="iframe-element" tabindex="0" class="container c2"></iframe>
-	<div tabindex="0" class="box b1" id="b1" style="left: 70px; top: -180px;"></div>
-	<div tabindex="0" class="box b1" id="b2" style="left: 180px; top: -120px;"></div>
+  <div tabindex="0" class="box b1" id="b1" style="left: 70px; top: 10px; margin-bottom: 20px;"></div>
+  <iframe src="iframe_sub.html" id="iframe-element" tabindex="0" class="container c2"></iframe>
```
Comment (jeonghee27) [inline tests/internal/iframe-test.html]: indentation

File: tests/internal/iframe-test.html [hunk 2]
```diff
@@ -10,11 +10,11 @@
...
 <body onload="onload()">
-	<iframe src="iframe_sub.html" id="iframe-element" tabindex="0" class="container c2"></iframe>
-	<div tabindex="0" class="box b1" id="b1" style="left: 70px; top: -180px;"></div>
-	<div tabindex="0" class="box b1" id="b2" style="left: 180px; top: -120px;"></div>
+  <div tabindex="0" class="box b1" id="b1" style="left: 70px; top: 10px; margin-bottom: 20px;"></div>
+  <iframe src="iframe_sub.html" id="iframe-element" tabindex="0" class="container c2"></iframe>	
+	<div tabindex="0" class="box b1" id="b2" style="left: 120px; top: -150px;"></div>
 	<div tabindex="0" class="box b1" id="b3" style="left: 20px; top: -90px;"></div>
-	<div tabindex="0" class="box b1" id="b4" style="left: 190px; top: -50px;"></div>
+	<div tabindex="0" class="box b1" id="b4" style="left: 190px; top: -20px;"></div>
```
Comment (jeonghee27) [inline tests/internal/iframe-test.html]: Why such position change is needed? 
when it comes to fixing a bug, Modifying the TC should be avoided.

Comment (jihyerish) [reply tests/internal/iframe-test.html]: Oh, I'll be careful about this next time. 
The reason for this is, I just simplified the test case and I thought the changing doesn't impact much about the test results.

**Comment by jihyerish** [comment]
> Ah.. I mean the front indentation.
> 
> has "tab" indentation, but <iframe src="iframe_sub.html" id="iframe-element" tabindex="0" class="container c2"></iframe> <iframe src="iframe_sub.html" id="iframe-element" tabindex="0" class="container c2"></iframe> have "2 space-bar" indentation.

Fix this with https://github.com/WICG/spatial-navigation/pull/201/commits/623ac1bbfe2a07c5bbe9abc9f800806ca2db5129

**Pull request merged**